ui/src/features/PageWrapper/WorkspaceId.tsx

```diff
@@ -0,0 +1,39 @@
+import { useParams } from "@tanstack/react-router";
+import { useEffect } from "react";
+
+import { Loading } from "@flow/components";
+import { useWorkspace } from "@flow/lib/gql";
+import { useCurrentWorkspace } from "@flow/stores";
+
+import NotFoundPage from "../NotFoundPage";
+
+type Props = {
+  children: React.ReactNode;
+};
+
+const WorkspaceIdWrapper: React.FC<Props> = ({ children }) => {
+  const [, setCurrentWorkspace] = useCurrentWorkspace();
+
+  const { workspaceId }: { workspaceId: string } = useParams({
+    strict: false,
+  });
+
+  const { useGetWorkspace } = useWorkspace();
+  const { workspace, isLoading } = useGetWorkspace(workspaceId);
+
+  useEffect(() => {
+    if (!workspace) return;
+    setCurrentWorkspace(workspace);
```

Comment:
We also don't want to reset global state if currentWorkspace.id === workspace.id